### PR TITLE
(fix): Update onChange function in obs date field

### PR DESCRIPTION
**Pull request by NethmiRodrigo**
## Requirements

- [x] This PR has a title that briefly describes the work done including the ticket number. If there is a ticket, make sure your PR title includes a [conventional commit](https://o3-docs.openmrs.org/docs/frontend-modules/contributing.en-US#contributing-guidelines) label. See existing PR titles for inspiration.
- [ ] My work conforms to the [OpenMRS 3.0 Styleguide](https://om.rs/styleguide) and [design documentation](https://om.rs/o3ui).
- [ ] My work includes tests or is validated by existing tests.

## Summary

This PR updates the onDateChange function in the obs date field that uses the `OpenmrsDatePicker`.

### CODE REVIEW

**Review by ibacher — APPROVED** [review]

**Comment by gracepotma** [comment]
@NethmiRodrigo what's the ticket # for this? Would like to ensure it's labelled as "Fix Version: 3.1.0" so we don't forget to include it in the next release.

File: packages/esm-patient-registration-app/src/patient-registration/field/obs/obs-field.component.tsx [hunk 1]
```diff
@@ -170,8 +171,8 @@ function DateObsField({ concept, label, required, placeholder }: DateObsFieldPro
   const { setFieldValue } = useContext(PatientRegistrationContext);
 
   const onDateChange = useCallback(
-    (date: Date) => {
-      setFieldValue(fieldName, date);
+    (birthdate: CalendarDate) => {
```
Comment (kajambiya) [inline packages/esm-patient-registration-app/src/patient-registration/field/obs/obs-field.component.tsx:174]: A small observation; this date can be any date so naming it birthdate is not fully representative of what it does.

**Review by kajambiya — COMMENTED** [review]
Hi @NethmiRodrigo 
This looks good, just left a quick comment.

File: packages/esm-patient-registration-app/src/patient-registration/field/obs/obs-field.component.tsx (hunk 1 above)
Comment (NethmiRodrigo) [reply packages/esm-patient-registration-app/src/patient-registration/field/obs/obs-field.component.tsx]: You are absolutely right, that was a grave error on my part. Thanks for catching it!

**Comment by NethmiRodrigo** [comment]
> @NethmiRodrigo what's the ticket # for this? Would like to ensure it's labelled as "Fix Version: 3.1.0" so we don't forget to include it in the next release.

@gracepotma very sorry Grace, I didn't file a ticket for this but maybe we could use [3482](https://openmrs.atlassian.net/browse/O3-3482)?

**Review by denniskigen — APPROVED** [review]
LGTM. Thanks, @NethmiRodrigo.

**Pull request merged**